File: app/src/main/java/com/app/test/data/network/HttpPageResponse.java
package com.app.test.data.network;

import com.app.test.data.entity.Page;

/**
 * <br/>
 * Created by Sweven on 2023/9/26--13:14.
 * Email: sweventears@163.com
 */
public class HttpPageResponse<T> extends Page<T> {
    private int code;

    public int getCode() {
        return code;
    }

    public void setCode(int code) {
        this.code = code;
    }
}

File: app/src/main/java/com/app/test/data/network/HttpPageResponse.kt
package com.app.test.data.network

/**
 * Created by Sweven on 2025/9/12--20:40.
 * Email: sweventears@163.com
 */
open class HttpPageResponse<T> @JvmOverloads constructor(
    var page: Int = 1,
    var lastPage: Int = 1,
    var pageSize: Int = 0,
    var total: Int = 0
) : HttpResponse<List<T>>()